File: test/fibonacci/fibonacci.c
#include "fibonacci.h"

int fibonacci(int n) {
    if (n <= 0) {
        return 0;
    }
    if (n == 1) {
        return 1;
    } else {
        int a = 0;
        int b = 1;
        int result = 0;
        for (int i = 2; i <= n; i++) {
            result = a + b;
            a = b;
            b = result;
        }
        return result;
    }
}
int main() {
    printf("Inserisci il numero di termini della serie di Fibonacci da calcolare:\n");
    float input = c_lua_io_read_number();
    if (input) {
        printf("Serie di Fibonacci:\n");
        for (int i = 0; i <= input - 1; i++) {
            printf("Termine %d : %d\n", i, fibonacci(i));
        }
    } else {
        printf("Per favore, inserisci un numero valido.\n");
    }
    return 0;
}

File: test/fibonacci/valid/fibonacci.lua
-- Programma per calcolare la serie di Fibonacci

function fibonacci(n)
    if n <= 0 then
        return 0
    end
    if n == 1 then
        return 1
    else
        a = 0
        b = 1
        result = 0
        for i = 2, n do
            result = a + b
            a = b
            b = result
        end
        return result
    end
end

print("\nEnter your name:")
user_name = io.read()
print("Hello, ", user_name)

print("Inserisci il numero di termini della serie di Fibonacci da calcolare:")
input = io.read("*n")

if input then
    print("Serie di Fibonacci:")
    for i = 0, input - 1 do
        print("Termine", i, ":", fibonacci(i))
    end
else
    print("Per favore, inserisci un numero valido.")
end